dataplane/saiserver/acl.go

```diff
@@ -292,3 +292,125 @@ func (a *acl) CreateAclEntry(ctx context.Context, req *saipb.CreateAclEntryReque
+					Bytes:   mi.macAddress,
+					Masks:   mi.macAddressMask,
+				}},
+			},
+		},
+	}
+	if withPriority {
+		ed.GetFlow().Priority = *mi.priority
+	}
+
+	if mi.vlanID != nil {
+		ed.GetFlow().Fields = append(
+			ed.GetFlow().Fields,
+			&fwdpb.PacketFieldMaskedBytes{
+				FieldId: &fwdpb.PacketFieldId{Field: &fwdpb.PacketField{FieldNum: fwdpb.PacketFieldNum_PACKET_FIELD_NUM_VLAN_TAG}},
+				Bytes:   binary.BigEndian.AppendUint32(nil, *mi.vlanID),
+				Masks:   binary.BigEndian.AppendUint32(nil, 0x00000FFF), // Match VLAN ID.
+			})
+	}
+
+	if mi.portID != nil {
+		fwdCtx, err := m.dataplane.FindContext(&fwdpb.ContextId{Id: m.dataplane.ID()})
+		if err != nil {
+			return nil, err
+		}
+		obj, err := fwdCtx.Objects.FindID(&fwdpb.ObjectId{Id: fmt.Sprint(*mi.portID)})
+		if err != nil {
+			return nil, err
+		}
+		ed.GetFlow().Fields = append(
+			ed.GetFlow().Fields,
+			&fwdpb.PacketFieldMaskedBytes{
+				FieldId: &fwdpb.PacketFieldId{Field: &fwdpb.PacketField{FieldNum: fwdpb.PacketFieldNum_PACKET_FIELD_NUM_PACKET_PORT_INPUT}},
+				Bytes:   binary.BigEndian.AppendUint64(nil, uint64(obj.NID())),
+				Masks:   binary.BigEndian.AppendUint64(nil, math.MaxUint64),
+			})
+	}
+	return ed, nil
+}
+
+type myMac struct {
+	saipb.UnimplementedMyMacServer
+	mgr        *attrmgr.AttrMgr
+	dataplane  switchDataplaneAPI
+	myMacTable map[uint64]*myMacInfo
+}
+
+func newMyMac(mgr *attrmgr.AttrMgr, dataplane switchDataplaneAPI, s *grpc.Server) *myMac {
+	m := &myMac{
+		mgr:        mgr,
+		dataplane:  dataplane,
+		myMacTable: map[uint64]*myMacInfo{},
+	}
+	saipb.RegisterMyMacServer(s, m)
+	return m
+}
+
+func (m *myMac) CreateMyMac(ctx context.Context, req *saipb.CreateMyMacRequest) (*saipb.CreateMyMacResponse, error) {
+	mi := &myMacInfo{
+		priority:       req.Priority,
+		portID:         req.PortId,
+		vlanID:         req.VlanId,
+		macAddress:     req.GetMacAddress(),
+		macAddressMask: req.GetMacAddressMask(),
+	}
+	if mi.macAddress == nil || mi.macAddressMask == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "MAC address and MAC address mask cannot be empty")
+	}
+	id := m.mgr.NextID()
+	ed, err := mi.ToEntryDesc(m, true /*with_priority*/)
+	if err != nil {
+		return nil, status.Errorf(codes.FailedPrecondition, "failed to create entry descriptor: %v", err)
+	}
+	mReq := &fwdpb.TableEntryAddRequest{
+		ContextId: &fwdpb.ContextId{Id: m.dataplane.ID()},
+		TableId:   &fwdpb.TableId{ObjectId: &fwdpb.ObjectId{Id: MyMacTable}},
+		EntryDesc: ed,
+	}
+
+	if _, err := m.dataplane.TableEntryAdd(ctx, mReq); err != nil {
+		return nil, err
+	}
+	return &saipb.CreateMyMacResponse{Oid: id}, nil
+}
+
+func (m *myMac) RemoveMyMac(ctx context.Context, req *saipb.RemoveMyMacRequest) (*saipb.RemoveMyMacResponse, error) {
+	mi, ok := m.myMacTable[req.GetOid()]
+	if !ok {
+		return nil, status.Errorf(codes.FailedPrecondition, "%d does not exist", req.GetOid())
+	}
+	ed, err := mi.ToEntryDesc(m, false /*with_priority*/)
```

Comment:
priority is required to delete the entry, remove this and always include it

see https://github.com/openconfig/lemming/blob/main/dataplane/forwarding/fwdtable/flow/flow.go#L236